libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.WebApi/BotFrameworkAdapterEx.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.WebApi
```

Comment:
The pleasure was all mine.

Reply:
Interesting... when I first looked at this I thought "Wait, woah, why is there another entire implementation of an adapter just for WebAPI?".

I really don't understand why the adapter model is effectively being pushed up the stack to the integration layers. This seems to corrupt the entire `BotAdapter` abstraction. Now there are two completely different `BotFrameworkAdapterEx` classes (and interfaces), each pegged to a specific integration. The whole point of the integration layers is to own the normalization of the incoming hosting layer representation to a representation that allows the underlying adapter to be agnostic of the host.

I'm honestly failing to see how this is a step forward at all as it takes what was one class (the existing `BotFrameworkAdapter`) and blows it into two and shoves host layer specifics inside of them rather than above it. Maybe I need to see some actual host integration code now for the picture to come together... unfortunately that's not in the PR yet.

Reply:
As a general statement the Adapter is the integration layer. You subclass the BotAdapter to add a specific protocol, that is the architecture. @benbrown actually makes quite a lot of usage of that idea.

The ASP integration was the outlier in that it can this significant additional moving part.

Interestingly this simple change made a big step to aligning with the JavaScript where @stevenic had always taken the webRequest object.